// @coinbase/onchainkit/dist/core/network/request.js
import { JSON_RPC_VERSION, RequestContext, JSON_HEADERS, CONTEXT_HEADER, POST_METHOD } from "./constants.js";
import { getRPCUrl } from "./getRPCUrl.js";
function buildRequestBody(method, params) {
  return {
    id: 1,
    jsonrpc: JSON_RPC_VERSION,
    method,
    params
  };
}
function buildRequestHeaders(context) {
  if (context) {
    if (!Object.values(RequestContext).includes(context)) {
      return {
        ...JSON_HEADERS,
        [CONTEXT_HEADER]: RequestContext.API
      };
    }
    return {
      ...JSON_HEADERS,
      [CONTEXT_HEADER]: context
    };
  }
  return JSON_HEADERS;
}
async function sendRequest(method, params, _context) {
  try {
    const body = buildRequestBody(method, params);
    const url = getRPCUrl();
    const response = await fetch(url, {
      body: JSON.stringify(body),
      headers: buildRequestHeaders(_context),
      method: POST_METHOD
    });
    const data = await response.json();
    return data;
  } catch (error) {
    console.log(
      `sendRequest: error sending request: ${error.message}`
    );
    throw error;
  }
}
export {
  buildRequestBody,
  buildRequestHeaders,
  sendRequest
};
